documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -547,6 +547,30 @@ but can be <<search-dsl-predicate-common-boost,boosted>>,
 either on a per-field basis with a call to `.boost(...)` just after `.field(...)`/`.fields(...)`
 or for the whole predicate with a call to `.boost(...)` after `.matching(...)`.
 
+[[search-dsl-predicate-regexp]]
+== `regexp`: match a regular expression pattern
+
+The `regexp` predicate matches documents for which a given field contains a word matching the given regular expression.
+
+.Matching a regular expression pattern
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=regexp]
+----
+====
+
+[IMPORTANT]
+====
+The provided regular expressions used in regexp predicates
+will never be normalized or transformed in any way,
+even if a normalizer/analyzer has been defined on the field.
+
+This means that if a normalizer/analyzer has been defined on the field,
+the user needs to be aware about how this normalizer/analyzer processes the data
+in order to provide a meaningful regular expression for using the predicate.
+====
+
```

Comment:
Please follow the same format as the documentation of other predicates. See for examples the sections "Targeting multiple fields" or "Other options" for the wildcard predicate.